src/cocotb/__init__.py

```diff
@@ -290,7 +290,13 @@ def _initialise_testbench_(argv_):
     top = cocotb.handle.SimHandle(handle)
 
     global regression_manager
-    regression_manager = RegressionManager.from_discovery(top)
+    try:
+        regression_manager = RegressionManager.from_discovery(top)
+    except BaseException as e:
+        log.error(e)
+        simulator.stop_simulator()
+        _stop_library_coverage()
```

Comment:
Is this redundant since _sim_event would be called?

Reply:
It is not redundant. `simulator.stop_simulator` deregisters the sim end callback (not sure why, this isn't a particularly good idea IMO), so the `_sim_event` callback won't be called otherwise.